core/utils/queue.h

```diff
@@ -0,0 +1,24 @@
+#ifndef BESS_UTILS_QUEUE_H
+#define BESS_UTILS_QUEUE_H
+
+namespace bess {
+namespace utils {
+
+// Takes Template argument T that is the type to be enqueued and dequeued.
+template <typename T>
+class Queue {
+ public:
+  virtual ~Queue(){};
+  virtual int Push(T*) = 0;
```

Comment:
In converting the Codel to be pointer independent, I ran into an issue for the Pop method where I can no longer return null when there are no objects(this breaks my current implementation). I have a few alternative implementation options: 1. return an object and take an error parameter to set, 2. return error parameter and take an object by reference to set, 3. throw an error when there are no objects left. none seem very clean although the last is more conventional for queues. I was really unsure on what to go with. thoughts?